### PR TITLE
fix(www): Eliminate horizontal scroll on mobile devices

### PR DESCRIPTION
**Pull request by qmixi**
## Description
 The purpose of this PR is to eliminate horizontal scroll on mobile devices cased by `egghead iframe videos` with hardcoded `600px` width attribute. This PR solves this issue and furthermore, makes egghead iframe videos responsive. 
The maximum width - 600px remains at it was before :) 

Screenshots below present the problem with a horizontal scroll and how it looks with the dynamic height/width calculation.

Without RWD support:
Screenshot 2019-10-21 at 22 34 24
Screenshot 2019-10-21 at 22 34 08
Screenshot 2019-10-21 at 22 33 47

With RWD support:
Screenshot 2019-10-21 at 22 24 11
Screenshot 2019-10-21 at 22 23 55
Screenshot 2019-10-21 at 22 23 45
Screenshot 2019-10-21 at 22 23 35
Screenshot 2019-10-21 at 22 23 26

I verified other pages that use the `egghead-embed.js` component and they work properly with the changes from PR as well.

### CODE REVIEW

**Comment by marcysutton** [comment]
Thanks for this! We need to test it on some real devices too, if you get a chance.

**Comment by qmixi** [comment]
> Thanks for this! We need to test it on some real devices too, if you get a chance.

Thank you! :) I run it on my iphone X and there is no longer horizontal scroll. Everything is working properly.
IMG_5800
IMG_5801

File: www/src/components/shared/egghead-embed.js [hunk 1]
```diff
@@ -1,25 +1,42 @@
...
-      </em>
-      .
-    </p>
-  </Fragment>
-)
+const EggheadEmbed = ({ lessonLink, lessonTitle }) => {
+  const [iframeWidth, setIframeWidth] = useState(0)
+  const iframeRef = useRef()
+
+  const handleResize = () => setIframeWidth(iframeRef.current.clientWidth)
```
Comment (igogo5yo) [inline www/src/components/shared/egghead-embed.js]: Hey, how about adding `useCallback` here?

Comment (igogo5yo) [reply www/src/components/shared/egghead-embed.js]: Thanks

File: www/src/components/shared/egghead-embed.js [hunk 2]
```diff
@@ -1,25 +1,45 @@
...
+    }
+  }, [iframeRef.current])
+
+  return (
+    <>
+      <iframe
+        ref={iframeRef}
+        className="egghead-video"
+        width={600}
+        height={iframeWidth * 0.57}
```
Comment (sidharthachatterjee) [inline www/src/components/shared/egghead-embed.js]: Why `0.57` specifically? 🤔 

I assume that's the aspect ratio of the videos generally? What if that changes?

Comment (qmixi) [reply www/src/components/shared/egghead-embed.js]: @sidharthachatterjee yes, it is the ratio of the height to the width of the video element inside the iframe, which is fixed. I think that it won't change as it's a native ratio of the video. Maybe it's a good idea to create a `constant` variable on the top of this component, so other users will understand what this number stands for? What do you think about this?

Comment (sidharthachatterjee) [reply www/src/components/shared/egghead-embed.js]: Yup, could definitely be a constant as opposed to a magic number

Comment (qmixi) [reply www/src/components/shared/egghead-embed.js]: @sidharthachatterjee commit with this improvement added :)

Comment (sidharthachatterjee) [reply www/src/components/shared/egghead-embed.js]: @qmixi Could also just be `0.5625` or `9/16` :)

File: www/src/components/shared/egghead-embed.js [hunk 3]
```diff
@@ -1,25 +1,47 @@
...
-
-    <p>
-      <em>
-        Video hosted on <a href={lessonLink}>egghead.io</a>
-      </em>
-      .
-    </p>
-  </Fragment>
-)
+const VIDEO_RATIO = 0.57
```
Comment (sidharthachatterjee) [inline www/src/components/shared/egghead-embed.js]:
```suggestion
const VIDEO_RATIO = 9/16
```

**Comment by fk** [comment]
Thank you @qmixi and @sidharthachatterjee! 🙏 💜 🤗

**Pull request merged**